File: Data Structures & Algorithms/contains-duplicate-ii/submission-3.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        l = -1
        numSet = defaultdict(int)
        for i, num in enumerate(nums):
            if num not in numSet:
                numSet[num] = i
            else:
                if i - numSet[num] <= k:
                    return True
                else:
                    numSet[num] = i
        return False

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if (nums.length === 0) return 0
        let set = new Set(nums);
        let maxLen = 1;
        let tempLen = 1;
        for (let num of nums) {
            if (!set.has(num)) continue
            set.delete(num)
            let tempNum = num;
            while (set.has(tempNum-1)) {
                tempLen++;
                tempNum--;
                set.delete(tempNum);
            }
            tempNum = num;
            while (set.has(tempNum+1)) {
                tempLen++;
                tempNum++;
                set.delete(tempNum);
            }
            maxLen = tempLen > maxLen ? tempLen : maxLen;
            tempLen = 1
        }
        return maxLen;
    }
}
